torch/csrc/api/include/torch/enum.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+
+#define TORCH_ENUM_DECLARE(name) \
+namespace torch { \
+namespace enumtype { \
+  /*
+    NOTE: We need to provide the default constructor for each struct,
+    otherwise Clang 3.8 would complain:
+    ```
+    error: default initialization of an object of const type 'const enumtype::Enum1'
+    without a user-provided default constructor
+    ```
+  */ \
+  struct k##name { k##name() {} }; \
+} \
+TORCH_API extern const enumtype::k##name k##name; \
+}
```

Comment:
Note to self: add test for torch.h include that should include all the enums. add a print test that prints the enum name given token (add operator<< for all enums)